### PR TITLE
Adding to the AMF "migration guide"

### PR DESCRIPTION
**Pull request by rddheath**
These revisions and additions to the content of the AMF "migration guide" are based on the Google Doc "Warnings and Errors in API Designer": https://docs.google.com/document/d/1eV9fI5CLX8qfzWgHyHdw98jAfXIjEVrC68lCE5D4piM/edit?ts=5c422bb7

In the files that are part of this PR, please search for this comment three times:

   `Review from here down`

This comment indicates where in the files to review. There are three files that contain this comment. Because of how I shuffled existing information around, I figured it would be easier for you to review if I pointed you directly at what to review.

There are two items from the Google Doc that I did not document:

- I had already documented Warning #2, working from the doc that Hernan Najles wrote up here: https://docs.google.com/document/d/1pJt_IjCdEC8CoPKs3T7RS4SzMdhtJMAeBmCTjd8nHNs/edit?ts=5c41e367#heading=h.jwx3x22asgfr

- I had already documented Error #6 by creating this file here, working from information provided to me by Antonio Garrote: https://docs.mulesoft.com/design-center/design-json-schema-required-error

### CODE REVIEW

File: modules/ROOT/pages/design-common-problems-raml-08-10.adoc [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+If an example value is required to be a `string`, but the value is instead an `integer`, then the editor displays this error message:
+
+----
+expecting !!str and !!int provided
+----
+
+For example, the example value for `a` must be in quotation marks for the editor to parse it as a `string`:
+----
+#%RAML 1.0
```
Comment (looseale) [inline modules/ROOT/pages/design-common-problems-raml-08-10.adoc]: Change the example to this two:

#%RAML 1.0
title: test
types:
  400:
    type: string

To

#%RAML 1.0
title: test
types:
  '400':
    type: string



And



#%RAML 1.0
title: test
types:
  A:
    properties:
      a: string
      b: integer
    examples:
      1:
        a: some
        b: 16
      2:
        a: other
        b: 32

To

#%RAML 1.0
title: test
types:
  A:
    properties:
      a: string
      b: integer
    examples:
      '1':
        a: some
        b: 16
      '2':
        a: other
        b: 32

File: modules/ROOT/pages/design-common-problems-raml-08-10.adoc [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+
+The editor would display this message for the following API specification:
+
+----
+#%RAML 1.0
+title: Example API Spec
+
+types:
+ User:
+   type: object
```
Comment (looseale) [inline modules/ROOT/pages/design-common-problems-raml-08-10.adoc]: Add the line "additionalProperties: false" down this line

File: modules/ROOT/pages/design-common-problems-raml-08-10.adoc [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+     firstName: string
+     lastName: string
+     age: integer
+   example:
+     firstName: John
+     lastName: Smith
+     age: 49
+----
+
+* Add the field `additionalProperties` with a value of `true` in the type declaration.
```
Comment (looseale) [inline modules/ROOT/pages/design-common-problems-raml-08-10.adoc]: Change this line to "Change the value of the field additionalProperties to true or remove it from the type declaration (by default additionalProperties is true)"

File: modules/ROOT/pages/design-common-problems-raml-08-10.adoc (hunk 1 above)
Comment (rddheath) [reply modules/ROOT/pages/design-common-problems-raml-08-10.adoc]: I've made the changes for this comment. Please have a look.

File: modules/ROOT/pages/design-common-problems-raml-08-10.adoc (hunk 2 above)
Comment (rddheath) [reply modules/ROOT/pages/design-common-problems-raml-08-10.adoc]: Resolved. Please have a look.

File: modules/ROOT/pages/design-common-problems-raml-08-10.adoc (hunk 3 above)
Comment (rddheath) [reply modules/ROOT/pages/design-common-problems-raml-08-10.adoc]: Resolved. Please have a look.

File: modules/ROOT/pages/design-common-problems-raml-08-10.adoc [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+
+== Not putting quotation marks around integers that are used as example values for strings
+
+If an example value is required to be a `string`, but the value is instead an `integer`, then the editor displays this error message:
+
+----
+expecting !!str and !!int provided
+----
+
+For example, the example value for `a` must be in quotation marks for the editor to parse it as a `string`:
```
[the remark this answers is not in the document]
Comment (rddheath) [reply modules/ROOT/pages/design-common-problems-raml-08-10.adoc]: I made different changes, explaining that the names of types are considered to be strings, so integers must be in quotation marks.

**Pull request merged**